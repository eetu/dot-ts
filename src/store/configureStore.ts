import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../reducers';

export default function configureStore(/*initialState: RootState*/) {
  const store = createStore(
    rootReducer,
    // initialState,
    applyMiddleware(thunk),
    // tslint:disable-next-line
    window['__REDUX_DEVTOOLS_EXTENSION__'] &&
      // tslint:disable-next-line
      window['__REDUX_DEVTOOLS_EXTENSION__'](),
  );

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const nextRootReducer = require('../reducers').default;
      store.replaceReducer(nextRootReducer);
    });
  }

  return store;
}
