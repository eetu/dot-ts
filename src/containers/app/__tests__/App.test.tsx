import * as React from 'react';
import createHistory from 'history/createBrowserHistory';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '../';
import configureStore from '../../../store/configureStore';

test('App should render', () => {

  const store = configureStore();
  const history = createHistory();

  const wrapper = shallow(
    <App store={store} history={history}/>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
