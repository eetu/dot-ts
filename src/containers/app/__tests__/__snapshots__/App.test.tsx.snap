// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render 1`] = `
<Translate(App)
  dot={
    Object {
      "dots": "",
    }
  }
  dotActions={
    Object {
      "add": [Function],
    }
  }
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  langActions={
    Object {
      "langEn": [Function],
      "langFi": [Function],
    }
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
